petab/math/SympyVisitor.py

```diff
@@ -0,0 +1,259 @@
+        return sp.Pow(operand1, operand2)
+
+    def visitUnaryExpr(self, ctx: PetabMathExprParser.UnaryExprContext):
+        if ctx.getChildCount() == 2:
+            operand = bool2num(self.visit(ctx.getChild(1)))
+            match ctx.getChild(0).getText():
+                case "-":
+                    return -operand
+                case "+":
+                    return operand
+        raise AssertionError(f"Unexpected expression: {ctx.getText()}")
+
+    def visitProg(self, ctx: PetabMathExprParser.ProgContext):
+        return self.visit(ctx.getChild(0))
+
+    def visitComparisonExpr(
+        self, ctx: PetabMathExprParser.ComparisonExprContext
+    ):
+        if ctx.getChildCount() != 3:
+            raise AssertionError(f"Unexpected expression: {ctx.getText()}")
+        lhs = self.visit(ctx.getChild(0))
+        op = ctx.getChild(1).getText()
+        rhs = self.visit(ctx.getChild(2))
+
+        ops = {
+            "==": sp.Equality,
+            "!=": sp.Unequality,
+            "<": sp.StrictLessThan,
+            ">": sp.StrictGreaterThan,
+            "<=": sp.LessThan,
+            ">=": sp.GreaterThan,
+        }
+        if op in ops:
+            lhs = bool2num(lhs)
+            rhs = bool2num(rhs)
+            return ops[op](lhs, rhs)
+        raise AssertionError(f"Unexpected operator: {op}")
+
+    def visitBooleanNotExpr(
+        self, ctx: PetabMathExprParser.BooleanNotExprContext
+    ):
+        if ctx.getChildCount() == 2:
+            return ~num2bool(self.visit(ctx.getChild(1)))
+
+        raise AssertionError(f"Unexpected expression: {ctx.getText()}")
+
+    def visitBooleanAndOrExpr(
+        self, ctx: PetabMathExprParser.BooleanAndOrExprContext
+    ):
+        if ctx.getChildCount() != 3:
+            raise AssertionError(f"Unexpected expression: {ctx.getText()}")
+
+        operand1 = num2bool(self.visit(ctx.getChild(0)))
+        operand2 = num2bool(self.visit(ctx.getChild(2)))
+
+        if ctx.BOOLEAN_AND():
+            return operand1 & operand2
+        if ctx.BOOLEAN_OR():
+            return operand1 | operand2
+
+        raise AssertionError(f"Unexpected expression: {ctx.getText()}")
+
+    def visitBooleanLiteral(
+        self, ctx: PetabMathExprParser.BooleanLiteralContext
+    ):
+        if ctx.TRUE():
+            return sp.true
+        if ctx.FALSE():
+            return sp.false
+        raise AssertionError(f"Unexpected boolean literal: {ctx.getText()}")
+
+
+def bool2num(x: sp.Basic):
+    """Convert sympy Booleans to Floats."""
+    if isinstance(x, BooleanFalse):
+        return sp.Float(0)
+    if isinstance(x, BooleanTrue):
+        return sp.Float(1)
+    return x
+
+
+def num2bool(x: sp.Basic):
+    """Convert sympy Floats to booleans."""
+    if isinstance(x, BooleanTrue | BooleanFalse):
+        return x
+    # Note: sp.Float(0) == 0 is False in sympy>=1.13
+    if x.is_zero is True:
+        return sp.false
+    if x.is_zero is False:
+        return sp.true
```

Comment:
```suggestion
    if x.is_zero in [True, False]:
        return not x.is_zero
```

Reply:
Here we want sympy's `true`, not python's `true`